lua/lualine.lua

```diff
@@ -65,17 +69,54 @@ local function load_special_components(component)
   end
 end
 
+local function component_loader(component)
+  if type(component[1]) == 'function' then return component end
+  -- apply default args
+  for opt_name, opt_val in pairs(M.opt) do
+    if component[opt_name] == nil then
+      component[opt_name] = opt_val
+    end
+  end
+  -- set custom highlights
+  if component.color then
```

Comment:
`color` is misleading. You are not really changing one color. Should be called `theme` imo. With this change there should also be an additional check here
```lua
if component.theme and component.theme ~= M.options.theme
```
I really like this idea overall

Reply:
>You are not really changing one color. Should be called theme imo.

We actualy might be changing just one color if user provides only fg or bg . I think theme is to broad an idea for this also probably will create confusion with M.theme . They are not the same theme . This option doesn't let you pick colors from other themes that many might think.

>```if component.theme and component.theme ~= M.options.theme```

I'm realy confused by this . What's it suppose to do?

Reply:
I think it's actually better with `theme`. Since user will have to remember only one variable name. I think component theme and component color are both perfectly readable. 

> ```
> if component.theme and component.theme ~= M.options.theme
> ```
> I'm realy confused by this . What's it suppose to do?

Well, if we rename it to theme then we will produce a slight issue. In line 78 we copy global options and if we try to update component's theme with global theme it will break

Reply:
>Well, if we rename it to theme then we will produce a slight issue. In line 78 we copy global options and if we try to update component's theme with global theme it will break

Well then that's another problem with theme .

I really donn't like the name theme . theme seems completely different than whats happening here and users will definatly expact theme names to work which wonn't

Reply:
And I'm not a fan of `color` but I guess I will let it slide for now. Let's keep `color`.